File: app/src/main/java/com/sendbird/calls/quickstart/fcm/MyFirebaseMessagingService.java
package com.sendbird.calls.quickstart.fcm;

import android.util.Log;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.sendbird.calls.SendBirdCall;
import com.sendbird.calls.quickstart.utils.PrefUtils;
import com.sendbird.calls.quickstart.utils.PushUtils;

public class MyFirebaseMessagingService extends FirebaseMessagingService {

    private static final String TAG = "MyFirebaseMessagingServ";

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        if (SendBirdCall.handleFirebaseMessageData(remoteMessage.getData())) {
            Log.d(TAG, "[SendBirdCall Message] onMessageReceived() => " + remoteMessage.getData().toString());
        } else {
            Log.d(TAG, "onMessageReceived() => From: " + remoteMessage.getFrom());
            if (remoteMessage.getData().size() > 0) {
                Log.d(TAG, "onMessageReceived() => Data: " + remoteMessage.getData().toString());
            }
            if (remoteMessage.getNotification() != null) {
                Log.d(TAG, "onMessageReceived() => Notification Body: " + remoteMessage.getNotification().getBody());
            }
        }
    }

    @Override
    public void onNewToken(@NonNull String token) {
        Log.d(TAG, "onNewToken(token: " + token + ")");

        if (SendBirdCall.getCurrentUser() != null)  {
            PushUtils.registerPushToken(getApplicationContext(), token, e -> {
                if (e != null) {
                    Log.d(TAG, "registerPushTokenForCurrentUser() => e: " + e.getMessage());
                }
            });
        } else {
            PrefUtils.setPushToken(getApplicationContext(), token);
        }
    }
}

File: directcall-kotlin/app/src/main/java/com/sendbird/calls/quickstart/fcm/MyFirebaseMessagingService.kt
package com.sendbird.calls.quickstart.fcm

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.sendbird.calls.SendBirdCall.currentUser
import com.sendbird.calls.SendBirdCall.handleFirebaseMessageData
import com.sendbird.calls.quickstart.TAG
import com.sendbird.calls.quickstart.utils.registerPushToken
import com.sendbird.calls.quickstart.utils.setPushToken

class MyFirebaseMessagingService : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        if (handleFirebaseMessageData(remoteMessage.data)) {
            Log.i(TAG, "[MyFirebaseMessagingService] onMessageReceived() => " + remoteMessage.data.toString())
        }
    }

    override fun onNewToken(token: String) {
        Log.i(TAG, "[MyFirebaseMessagingService] onNewToken(token: $token)")
        if (currentUser != null) {
            registerPushToken(token) {
                if (it != null) {
                    Log.i(TAG,"[MyFirebaseMessagingService] registerPushTokenForCurrentUser() => e: " + it.message)
                }
            }
        } else {
            applicationContext.setPushToken(token)
        }
    }
}
